six/seven_ans.py

```diff
@@ -0,0 +1,11 @@
+phone_dict = {'1':[''], '2':['A', 'B', 'C'], '3':['D', 'E', 'F'], '4':['G', 'H', 'I'], '5':['J', 'K', 'L'], '6':['M', 'N', 'O'], '7':['P', 'Q', 'R', 'S'], '8':['T', 'U', 'V'], '9':['W', 'X', 'Y', 'Z'], '0':['']}
+
+def nieoh_num_to_char(num):
+    if len(num)==1:
+        return phone_dict[num[0]]
+    ret = []
+    num = list(num)
+    for char in phone_dict[num[-1]]:
+        ret += [n + char for n in nieoh_num_to_char(num[:-1])]
```

Comment:
by `+`ing different lists together you're technically doing unnecessary copying. Consider either doing `x.append(y)` inside 2 for loops or doing a more complicated list comprehension.